### PR TITLE
Android feature minimap

### PR DESCRIPTION
**Pull request by michaelzangl**
Currently it displays a random image. Can be fixed once #337 is merged.

### CODE REVIEW

File: jsettlers.graphics/src/jsettlers/graphics/map/minimap/Minimap.java [hunk 1]
```diff
@@ -269,7 +270,7 @@ public void setBufferArray(short[][] buffer) {
 	public void stop() {
 		lineLoader.stop();
 		stopped = true;
-		synchronized (update_syncobj) {
+		synchronized (updateMutex) {
```
Comment (andreas-eberle) [inline jsettlers.graphics/src/jsettlers/graphics/map/minimap/Minimap.java]: what is this for at this position? Nothing is in the mutex...

**Comment by andreas-eberle** [comment]
@michaelzangl: Can you please update this branch so I can merge it?

File: jsettlers.graphics/src/jsettlers/graphics/map/minimap/Minimap.java (hunk 1 above)
Comment (michaelzangl) [inline jsettlers.graphics/src/jsettlers/graphics/map/minimap/Minimap.java]: We simply wait if the mutex is currently locked until it gets unlocked.

**Comment by michaelzangl** [comment]
I'd like to wait until after the gradle merge with this.

File: jsettlers.graphics.androidui/src/main/java/jsettlers/graphics/androidui/menu/MinimapMenu.java [hunk 2]
```diff
@@ -14,13 +14,186 @@
...
+ * 
+ * @author Michael Zangl
+ */
 public class MinimapMenu extends AndroidMenu {
 
-	public MinimapMenu(AndroidMenuPutable puttable) {
-		super(puttable);
-		Minimap minimap = new Minimap(puttable.getMapContext(), new MinimapMode());
+	private final AndroidLineLoader loader = new AndroidLineLoader(new MinimapData(), new MinimapMode());
+	private MinimapView map;
```
Comment (andreas-eberle) [inline jsettlers.graphics.androidui/src/main/java/jsettlers/graphics/androidui/menu/MinimapMenu.java]: better call this minimapView

File: jsettlers.graphics.androidui/src/main/java/jsettlers/graphics/androidui/menu/MinimapMenu.java [hunk 3]
```diff
@@ -14,13 +14,186 @@
...
+		}
+
+		@Override
+		public MapDrawContext getContext() {
+			return getPutable().getMapContext();
+		}
+
+		@Override
+		public void blockUntilUpdateAllowedOrStopped() {
+			System.out.println("Minimap loop.");
```
Comment (andreas-eberle) [inline jsettlers.graphics.androidui/src/main/java/jsettlers/graphics/androidui/menu/MinimapMenu.java]: remove debug output please.

File: jsettlers.graphics.androidui/src/main/java/jsettlers/graphics/androidui/menu/MinimapView.java [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+	}
+
+	@Override
+	protected void onSizeChanged(int w, int h, int oldw, int oldh) {
+		super.onSizeChanged(w, h, oldw, oldh);
+
+		if (bitmap != null) {
+			matrix = new Matrix();
+			float s = Math.min((w - 20f) / bitmap.getWidth() / (1 + SKEW), (h - 20f) / bitmap.getHeight());
+			System.out.println(s);
```
Comment (andreas-eberle) [inline jsettlers.graphics.androidui/src/main/java/jsettlers/graphics/androidui/menu/MinimapView.java]: remove debug output please

**Comment by andreas-eberle** [comment]
The minimap looks a bit different on Android compared to PC. Its angle is steeper. See the two screenshots.

Android
screenshot_20160424-211113

PC
unbenannt

File: jsettlers.graphics.androidui/src/main/java/jsettlers/graphics/androidui/hud/HudFragment.java [hunk 5]
```diff
@@ -60,4 +78,9 @@ public void onStop() {
 		getPutable().getChangeObserveable().removeMapSelectionListener(selectionSetter);
 		super.onStop();
 	}
+
+	public void gameStop() {
+		//TODO: Call this.
```
Comment (andreas-eberle) [inline jsettlers.graphics.androidui/src/main/java/jsettlers/graphics/androidui/hud/HudFragment.java]: is this done? If so, we should remove the todo.

File: jsettlers.graphics.androidui/src/main/java/jsettlers/graphics/androidui/menu/MinimapMenu.java [hunk 6]
```diff
@@ -14,13 +14,186 @@
...
+			// post() may be delayed, check if still necessary.
+			if (updateRunnerWanted) {
+				updateBitmap();
+			}
+		}
+	};
+
+	public MinimapMenu(AndroidMenuPutable putable) {
+		super(putable);
+		// TODO: State handling
```
Comment (andreas-eberle) [inline jsettlers.graphics.androidui/src/main/java/jsettlers/graphics/androidui/menu/MinimapMenu.java]: Should this be implemented now?

Comment (michaelzangl) [inline jsettlers.graphics.androidui/src/main/java/jsettlers/graphics/androidui/menu/MinimapMenu.java]: No. Thread stopping is missing

**Comment by michaelzangl** [comment]
The main problem with the minimap is that it does not resize if the aspect of the map changes. This is true for both versions.

**Comment by andreas-eberle** [comment]
Yes, this is a general problem, which we might be able to fix easier on Android (as we have our own UI and aren't limited to the form factor of the original). However, the circles of the towers should look like circles instead of like an ellipse.

**Pull request merged**